### ISSUE TITLE
Logstash install fails due to dependency on PGP server

### ISSUE THREAD

**Issue by peterskim12** [elastic/puppet-logstash#191]
When trying to install Logstash via Puppet module, I got this error:

==> default: Error: Execution of '/usr/bin/apt-key adv --keyserver pgp.mit.edu --recv-keys D88E42B4' returned 2: Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --homedir /tmp/tmp.gZdi1xiGZ2 --no-auto-check-trustdb --trust-model always --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyserver pgp.mit.edu --recv-keys D88E42B4
==> default: gpg: requesting key D88E42B4 from hkp server pgp.mit.edu
==> default: gpgkeys: key D88E42B4 can't be retrieved
==> default: gpg: no valid OpenPGP data found.
==> default: gpg: Total number processed: 0
==> default: Notice: /Stage[main]/Logstash::Repo/Apt::Source[logstash]/Apt::Key[Add key: D88E42B4 from Apt::Source logstash]/Anchor[apt_key D88E42B4 present]: Dependency Apt_key[Add key: D88E42B4 from Apt::Source logstash] has failures: true

Presumably the PGP service was down. @electrical mentioned that this should be changed to use a key file instead to cut off the dependency on the server being up.

**Comment by cyrilzloi** [comment]
Works for me under Debian 7.6:

"key_server  => 'hkp://pgp.mit.edu:80',"   in repo.pp manifest.

**Comment by mikey-** [comment]
Hey guys ^_^ other than editing key_server, is there any kind of work around for this?

**Comment by mikey-** [comment]
[I fixed this here] (https://github.com/mikey-/puppet-logstash/blob/bugfix/191-use-GPG-key/manifests/repo.pp#L44). I'd PR but I couldn't get the rspec tests to run (before I made any change). Anyway, Ima just use this fork for a while i guess <3 <3

**Comment by cyrilzloi** [comment]
Nice fix, thanks. Ill test it)

**Comment by agonen** [comment]
I've tried to give it a test . got this error : 

==> default: Error: validate_re(): "46095ACC8548582C1A2699A9D27D666CD88E42B4" does not match ["\\A(0x)?[0-9a-fA-F]{8}\\Z", "\\A(0x)?[0-9a-fA-F]{16}\\Z"] at /etc/puppet/modules/apt/manifests/key.pp:60 on node vagrant-ubuntu-trusty-64
==> default: Error: validate_re(): "46095ACC8548582C1A2699A9D27D666CD88E42B4" does not match ["\\A(0x)?[0-9a-fA-F]{8}\\Z", "\\A(0x)?[0-9a-fA-F]{16}\\Z"] at /etc/puppet/modules/apt/manifests/key.pp:60 on node vagrant-ubuntu-trusty-64

I've Ubuntu 14.10

**Comment by mikey-** [comment]
Sorry for the late reply, should be more clear about this, the fix isn't just changing the key.

it's [changing the key] (https://github.com/mikey-/puppet-logstash/blob/bugfix/191-use-GPG-key/manifests/repo.pp#L44)
and [changing using key_source instead of key_server] (https://github.com/mikey-/puppet-logstash/blob/bugfix/191-use-GPG-key/manifests/repo.pp#L45)

hope that helps :s 

You should be able to use the 40 hex character long key fingerprint. do you know which version of the apt module you're using?

**Comment by electrical** [comment]
I did fix this issue in the Elasticsearch module but is planned to be ported over to this module as well in the very near future.

**Comment by mikey-** [comment]
nice, thanks! can't wait ;) <3

**Comment by AtzeDeVries** [comment]
very nice, but i still can't wait ;)

**Comment by jmkgreen** [comment]
As a result of other modules requiring the newer apt source module, we now find this module causing Puppet runs to fail:

```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Invalid parameter keyserver_options on Apt_key[Add key: D88E42B4 from Apt::Source logstash] at /etc/puppet/modules/apt/manifests/key.pp:95 on node virt-04.int.company.net
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

This would now seem rather urgent, to us at least.

**Comment by bflad** [comment]
Hi, interestingly you may be experiencing what we saw in our environment until _all_ nodes were running puppetlabs/apt >=2. See this for more details: https://github.com/jfryman/puppet-nginx/issues/650 (issue is not nginx specific). Basically, if you're running puppetserver, you'll need to restart the server or run the agent multiple times for a successful convergence until all nodes are upgraded.

**Comment by jmkgreen** [comment]
@bflad which makes things all the more absurd: trying puppetlabs/apt >= 2 in the "test" environment ahead of production roll-out - essentially production breaks test. At this point I'm unsure who to point the finger of blame at most but I'm sure having a bad experience one way or another.

**Comment by jmkgreen** [comment]
Manually patching this into our production logstash module with apt 1.8 gets us green again. I have dropped the 2.1 apt from our test rig just to get consistency again.

**Comment by jarpy** [comment]
@abednarik fixed this one in 65e175d65aebe2118b7b161924477e6c094b3a2d. :100:
